api/app/src/external/commonwell/document/shared.ts

```diff
@@ -1,39 +1,84 @@
-import { Document, DocumentContent } from "@metriport/commonwell-sdk";
+import { Document } from "@metriport/commonwell-sdk";
 import mime from "mime-types";
-import { MedicalDataSource } from "../..";
-import { DocumentReferenceCreate } from "../../../domain/medical/document-reference";
+import { DocumentReference } from "@medplum/fhirtypes";
 import { Patient } from "../../../models/medical/patient";
+import { Organization } from "../../../models/medical/organization";
 import { makePatientOID } from "../../../shared/oid";
+import { MedicalDataSource } from "../..";
 
 // TODO #340 When we fix tsconfig on CW SDK we can remove the `Required` for `id`
-export type DocumentWithLocation = Required<Pick<Document, "id">> &
-  Omit<DocumentContent, "location"> &
-  Required<Pick<DocumentContent, "location">> & {
-    fileName: string;
-    raw?: unknown;
-  };
+export type DocumentWithFilename = Document & {
```

Comment:
I thought because we had it on the `Document` it would infer that it is required? Will this be fixed by tsconfig if not then ill make it required

Reply:
Based on the comment there, I think we were not exposing the type properly from the package, so we had to add the `<Required>` there.

Also, the previous type was flattening the document, so there was no `content` on `DocumentWithLocation`. I don't know if that would still make sense with the changes on this PR, but previously I thought it made sense to simplify mapping to the local type of document references on the API (slightly diff from the one from CW SDK).

Reply:
I merged your PR and id is now being flagged as required, as for flattening it makes more sense to leave content as is because thats how its being added when we convert to FHIR